sql/sql_explain.cc

```diff
@@ -256,44 +259,75 @@ int Explain_query::print_explain_json(select_result_sink *output,
 #endif
 
   writer.start_object();
-  if (is_analyze && query_time_in_progress_ms > 0)
-    writer.add_member("r_query_time_in_progress_ms").
-           add_ull(query_time_in_progress_ms);
-
+  
   /*
     If we are printing ANALYZE FORMAT=JSON output, take into account that
     query's temporary tables have already been freed. See sql_explain.h,
     sql_explain.h:ExplainDataStructureLifetime for details.
   */
   if (is_analyze)
-    is_show_cmd= true;
+  {
+    if (query_time_in_progress_ms > 0){
+      writer.add_member("r_query_time_in_progress_ms").
+            add_ull(query_time_in_progress_ms);
+    }
+
+    print_query_optimization_json(&writer);
+    is_show_cmd = true;
+  }
+
+  bool were_query_plans_found = print_query_blocks_json(&writer, is_analyze, is_show_cmd);
```

Comment:
Also a very long name. Please change to `plan_found`  (it's singular. there can't be more than one).